swap/protocol_test.go

```diff
@@ -296,3 +303,156 @@ func TestTriggerDisconnectThreshold(t *testing.T) {
 		t.Fatalf("Expected still no cheque, but there are %d", lenCheques)
 	}
 }
+
+// TestSwapRPC tests some basic things over RPC
+// We want this so that we can check the API works
+func TestSwapRPC(t *testing.T) {
+
+	var (
+		p2pPort = 30100
+		ipcPath = ".swarm.ipc"
+		err     error
+	)
+
+	swap, clean := newTestSwap(t, ownerKey)
+	defer clean()
+
+	// need to have a dummy contract or the call will fail at `GetParams` due to `NewAPI`
+	swap.contract, err = contract.InstanceAt(common.Address{}, swap.backend)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// start a service stack
+	stack := createAndStartSvcNode(swap, ipcPath, p2pPort, t)
+	defer stack.Stop()
```

Comment:
This test spends most fo the time in stopping the node here, and since it does not validate rpc node lifecycle, but only swap balance calls, I think that it can be made "faster" with putting that in the background:

```go
	defer func() {
		go stack.Stop()
	}()
```

Reply:
booah waaaaaayyy fasterrrrrrr!!
;)

Is it ensured though that `Stop` will be executed, and that the test will not terminate before that, creating a memory leak?

Reply:
Well, test will terminate before Stop is done. There are two cases after the test is done. One is that there are no more tests to be executed (depending on go test command options or order of tests in the package) and the goroutine that calls Stop will be terminated before the Stop is done. This is ok, as the main goroutine is terminated also. Another case is that there are other tests, and that Stop goroutine will be scheduled somewhere in the future, but we do not know when. This should be also ok, as the stack is not a global variable, the test result is not dependent on Stop being called and we know that the stop will be done in some reasonable time, not creating a gorotuine leak. I think that it is ok to do this trick in this test.